Gems/MotionMatching/Code/Source/FeatureMatrixMinMaxScaler.cpp

```diff
@@ -0,0 +1,215 @@
+        }
+
+        return result;
+    }
+
+    //-------------------------------------------------------------------------
+
+    FeatureMatrix MinMaxScaler::InverseTransform(const FeatureMatrix& featureMatrix) const
+    {
+        const FeatureMatrix::Index numRows = featureMatrix.rows();
+        const FeatureMatrix::Index numColumns = featureMatrix.cols();
+        FeatureMatrix result;
+        result.resize(numRows, numColumns);
+
+        for (FeatureMatrix::Index row = 0; row < numRows; ++row)
+        {
+            for (FeatureMatrix::Index column = 0; column < numColumns; ++column)
+            {
+                result(row, column) = InverseTransform(featureMatrix(row, column), column);
+            }
+        }
+
+        return result;
+    }
+
+    AZ::Vector2 MinMaxScaler::InverseTransform(const AZ::Vector2& value, FeatureMatrix::Index column) const
+    {
+        return AZ::Vector2(
+            InverseTransform(value.GetX(), column + 0),
+            InverseTransform(value.GetY(), column + 1));
+    }
+
+    AZ::Vector3 MinMaxScaler::InverseTransform(const AZ::Vector3& value, FeatureMatrix::Index column) const
+    {
+        return AZ::Vector3(
+            InverseTransform(value.GetX(), column + 0),
+            InverseTransform(value.GetY(), column + 1),
+            InverseTransform(value.GetZ(), column + 2));
+    }
+
+    float MinMaxScaler::InverseTransform(float value, FeatureMatrix::Index column) const
+    {
+        const float normalizedValue = (value - m_featureMin) / m_featureRange;
+        return normalizedValue * m_dataRange[column] + m_dataMin[column];
+    }
+
+    void MinMaxScaler::SaveMinMaxAsCsv(const AZStd::string& filename, const AZStd::vector<AZStd::string>& columnNames)
+    {
+        AZStd::string data;
+
+        // Save column names in the first row
+        if (!columnNames.empty())
+        {
+            for (size_t i = 0; i < columnNames.size(); ++i)
+            {
+                if (i != 0)
+                {
+                    data += ",";
+                }
+
+                data += columnNames[i].c_str();
+            }
+            data += "\n";
+        }
+
+        for (size_t i = 0; i < m_dataMin.size(); ++i)
+        {
+            if (i != 0)
+            {
+                data += ",";
+            }
+
+            data += AZStd::to_string(m_dataMin[i]);
+        }
+
+        data += "\n";
+
+        for (size_t i = 0; i < m_dataMax.size(); ++i)
+        {
+            if (i != 0)
+            {
+                data += ",";
+            }
+
+            data += AZStd::to_string(m_dataMax[i]);
+        }
+
+        data += "\n";
+
+        AZ::IO::FileIOBase::SetInstance(aznew AZ::IO::LocalFileIO());
+        AZ::IO::FileIOStream file;
+        if (file.Open(filename.c_str(), AZ::IO::OpenMode::ModeWrite | AZ::IO::OpenMode::ModeText | AZ::IO::OpenMode::ModeCreatePath))
+        {
+            file.Write(data.size(), data.data());
+        }
+        file.Close();
```

Comment:
This shouldn't make a new FileIO instance.
This will [assert](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/IO/FileIO.cpp#L53) when used in actual O3DE applications that set a FileIO Instance, as a new FileIOBase instance cannot be set without setting the existing one to nullptr.

I advise using an AZ::IO::SystemFileStream here instead

Reply:
Thanks, I've changed it to use a AZ::IO::SystemFile